global_flags/global_flags.rs

```diff
@@ -0,0 +1,15 @@
+use lazy_static::lazy_static;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+
+lazy_static! {
+  pub static ref RUNNING_FROM_BINARY: AtomicBool = AtomicBool::new(false);
+}
+
+pub fn set_running_from_binary(value: bool) {
+  RUNNING_FROM_BINARY.store(value, Ordering::SeqCst);
+}
+
+pub fn is_running_from_binary() -> bool {
+  RUNNING_FROM_BINARY.load(Ordering::SeqCst)
+}
```

Comment:
I'm really not a fan of adding a separate crate just to share a single boolean. I think we can leverage other approaches here. 

I think we should use `OpState` here and add two "wrapper" structs: one in `ext/fs` and other in `runtime`. These struct should looks something like this:
```
struct IsStandaloneBinary(bool)
```
The we should accept these struct in `options` bag in `deno_fs` and `deno_runtime` extensions and look them up when deciding which message to print.

@yazan-abdalrahman let me know if this is enough info to get you started

Reply:
Hi @bartlomieju , Okay, I think it's enough. I'll edit it and include your notes.

Reply:
@bartlomieju 
I believe we can't use OpState because it is intended to maintain the resources and operations within a JS runtime.    and we need to provide a Boolean value to use it in chose prompt message and we don't know which code flow it will take.
So, we need it to be global, and I assumed in the future we would need something else in the same way, so I utilized crate.

and what is bag in **options bag in deno_fs and deno_runtime**?

Reply:
> I believe we can't use OpState because it is intended to maintain the resources and operations within a JS runtime.

You can use it for whatever purpose you need, `OpState` will work here perfectly fine.

> and we need to provide a Boolean value to use it in chose prompt message and we don't know which code flow it will follow.

I'm not sure what you mean, can you explain?

> So, we need it to be global, and I assumed in the future we would need something else in the same way, so I utilized crate.

It doesn't need to be global, it just needs to be available to permission crate. We didn't have a need to any global prior so I don't think this assumption is necessarily correct.

> and what is bag in options bag in deno_fs and deno_runtime?

These options:
https://github.com/denoland/deno/blob/3be0a1e8b44782b2d3c3970e0e6fa7fb11be8131/ext/fs/lib.rs#L253-L258
https://github.com/denoland/deno/blob/3be0a1e8b44782b2d3c3970e0e6fa7fb11be8131/runtime/worker.rs#L329-L337